src/lib/lore.sh

```diff
@@ -458,3 +464,152 @@ function download_series()
+
+  patch_file_name=$(convert_title_to_patch_name "$patch_title")
+
+  file_pattern=$(printf '%s/*%s' "${download_dir_path}" "${patch_file_name}")
+
+  ls "${file_pattern}" 1> /dev/null 2>&1
+
+  if [[ "$ret" == 0 ]]; then
+    command="rm ${file_pattern}"
+    cmd_manager "$flag" "$command"
+  else
+    return 2 # ENOENT
+  fi
+}
+
+# This function creates the lore bookmarked series file if it doesn't
+# already exists
+function create_lore_bookmarked_file()
+{
+  if [[ ! -f "${BOOKMARKED_SERIES_PATH}" ]]; then
+    touch "${BOOKMARKED_SERIES_PATH}"
+  fi
+}
+
+# This function adds a given series to a local bookmark database for
+# series managed by kw. To achieve uniqueness for each series the function
+# assigns an id to it. Also a timestamp for informational purposes is also
+# added.
+#
+# @target_patch: Patch (series) in the same format as list_of_mailinglist_patches
+#   to be added to local bookmark database
+# @download_dir_path: The directory where the patch (series) was saved
+function add_series_to_bookmark()
+{
+  local target_patch="$1"
+  local download_dir_path="$2"
+  local patch_id
+  local timestamp
+
+  create_lore_bookmarked_file
+
+  patch_id=$(printf '%s' "${target_patch}" | sha256sum | cut -d ' ' -f1)
+  timestamp=$(date '+%Y/%m/%d %H:%M')
+
+  if [[ $(grep -c "^${patch_id}" "${BOOKMARKED_SERIES_PATH}") == 0 ]]; then
+    {
+      printf '%s%s' "${patch_id}" "${SEPARATOR_CHAR}"
+      printf '%s%s' "$timestamp" "${SEPARATOR_CHAR}"
+      printf '%s%s' "${target_patch}" "${SEPARATOR_CHAR}"
+      printf '%s\n' "${download_dir_path}"
+    } >> "${BOOKMARKED_SERIES_PATH}"
+  fi
+}
+
+# This function removes a series from the local bookmark database by its index
+# in the database.
+#
+# @series_index: The index in the local bookmark database
+#
+# TODO:
+# - Find an alternative way to identify a series, this one may not be the most
+#   reliable.
+function remove_series_from_bookmark_by_index()
+{
+  local series_index="$1"
+
+  if [[ ! -f "${BOOKMARKED_SERIES_PATH}" ]]; then
+    return 2 # ENOENT
+  fi
+
+  sed -i "${series_index}d" "${BOOKMARKED_SERIES_PATH}"
+}
+
+# This function populates an array passed as argument with all the bookmarked
+# series. Each element will detain the information to be displayed in the bookmarked
+# patches screen.
+#
+# @_bookmarked_series: An array reference to be populated with all the bookmarked
+#   series.
+#
+# TODO:
+# - Better decide which information will be shown in the bookmarked patches screen
+function get_bookmarked_series()
+{
+  local -n _bookmarked_series="$1"
+  local index=0
+  local timestamp
+  local patch_title
+  local patch_author
+  local tmp_data
+
+  if [[ ! -f "${BOOKMARKED_SERIES_PATH}" ]]; then
+    return 2 # ENOENT
+  fi
+
+  _bookmarked_series=()
+
+  while IFS='' read -r series; do
+    IFS="${SEPARATOR_CHAR}" read -r -a columns <<< "$series"
+    timestamp="${columns[1]}"
+    patch_title="${columns[6]}"
+    patch_author="${columns[2]}"
```

Comment:
At some point, we need to replace this `${columns[1]}` in the entire code with something more meaningful. Using an index is really confusing... maybe we can have a set of functions for that?

Reply:
You are right. Even more considering that even in this same file they don't follow a pattern...

By a set a functions you are talking about something like `get_timestamp`, `get_patch_title` and so on to encapsulate this explicit `${columns[i]}`, right?

Reply:
For sure. The issue regarding the integration BD. Going to implement these functions.